File: crates/nil-payload/src/ranking.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use nil_core::ruler::Ruler;
use nil_core::world::WorldId;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetRankingRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetRankRequest {
  pub world: WorldId,
  pub ruler: Ruler,
}

File: app/src/types/request/ranking.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import type { Ruler } from '@/types/core/ruler';
import type { WorldId } from '@/types/core/world';

export interface GetRankingRequest {
  readonly world: WorldId;
}

export interface GetRankRequest {
  readonly world: WorldId;
  readonly ruler: Ruler;
}
